text/XXX-A-Compilation-of-Short-Interactive-Tutorials-and-How-To-Guides-for-Coq.md

```diff
@@ -0,0 +1,257 @@
+
+
+### 4.4 Online Interface
+
+We have identified 3 different components which play a role in making nice interactive web tutorials:
+- First, we need a tool to format structured Coq comments (coqdoc, embedded `rst`, ...) into `html`. This tool and the associated format for comments should be easy to use to encourage more people to contribute.
+- Second, we need a way to make Coq run inside a standard compliant web browser.
+- Finally, we need a nice user interface inside the interactive web pages which will contain different views: the proof state, Coq's messages, possibly a Search view, ...
+
+We already have some technical answers:
+- [Alectryon](https://github.com/cpitclaudel/alectryon) provides means to interpret restructured text inside Coq files (and conversely).
+- `jsCoq` and `waCoq` enable Coq to be executed inside a web browser, so there is a priori no insurmountable difficulty there.
+
+Yet, it will require some real technical work:
+- waCoq or jsCoq need to evolve to more recent versions of Coq and their usage with external libraries needs to be smoother,
+- we have not tested yet how to use Alectryon and jsCoq/waCoq together.
+
+Having such an easy to use online interface is of general interest for the Coq environment, e.g. teachers, so we hope they will be addressed soon.
+
+### 4.5 Moving on to the official Coq repository and Coq's webpage
+If this CEP is accepted, after discussion and amendments, we wish this documentation project to be part of the roadmap to gain visibility, and for the temporary repository to be moved to an official Coq organization repository.
+
+While discussing with some developers, the idea also emerged for this documentation project to be integrated directly to Rocq's new website, similarly to what is done on [Ocaml's website](https://ocaml.org/docs/lists). This is also a question to address with this CEP.
+
+
+### 4.6 Human resources
+
+This project will necessarily have to be collaborative considering the amount of work to be done, and the richness and diversity of our ecosystem. No one has the time nor is competent to write or review expertly every single possible tutorial. Yet, as the tutorials can mostly be designed independently, by combining the different expertise of the different communities, there is good hope to quickly get to a good documentation, and we welcome contributions. We hope to attract more and more contributors and involve plugin writers and Coq developers in this process.
+
+Yet, this project will require specific resources:
+- As mentioned in section 4.3, there is real work to be done on the interface side.
+- There is a need for someone interacting with the community, reviewing etc.
+
+Thomas Lamiaux should be employed part-time on this project as a 3 years expertise mission included in his PhD. His task should be developing the project, writing and reviewing tutorials, and interacting with the community, ensuring at least some people-power in the beginning of this project.
```

Comment:
Is this effort an integral part of your PhD work?  If so, what is your planned PhD work (just curious).

Reply:
It is not, feel free to check my webpage